test/integration/conf.go

```diff
@@ -127,8 +127,6 @@ func GetConfig(datapath string, configType uint32) pfcpiface.Conf {
 	}
 
 	panic("Wrong datapath or config type provided")
```

Comment:
should we return some error instead than panic?

Reply:
This could be a way to avoid the panic. What do you think?
```diff
diff --git a/test/integration/conf.go b/test/integration/conf.go
index a6c0d62..e6d3165 100644
--- a/test/integration/conf.go
+++ b/test/integration/conf.go
@@ -6,6 +6,7 @@ package integration
 import (
        "bytes"
        "encoding/json"
+       "fmt"
        "github.com/omec-project/upf-epc/pfcpiface"
        "github.com/sirupsen/logrus"
        "net/http"
@@ -106,27 +107,27 @@ func UP4ConfigWipeOutOnUP4Restart() pfcpiface.Conf {
        return config
 }
 
-func GetConfig(datapath string, configType uint32) pfcpiface.Conf {
+func GetConfig(datapath string, configType uint32) (pfcpiface.Conf, error) {
        switch datapath {
        case DatapathUP4:
                switch configType {
                case ConfigDefault:
-                       return UP4ConfigDefault()
+                       return UP4ConfigDefault(), nil
                case ConfigUPFBasedIPAllocation:
-                       return UP4ConfigUPFBasedIPAllocation()
+                       return UP4ConfigUPFBasedIPAllocation(), nil
                case ConfigWipeOutOnUP4Restart:
-                       return UP4ConfigWipeOutOnUP4Restart()
+                       return UP4ConfigWipeOutOnUP4Restart(), nil
                }
        case DatapathBESS:
                switch configType {
                case ConfigDefault:
-                       return BESSConfigDefault()
+                       return BESSConfigDefault(), nil
                case ConfigUPFBasedIPAllocation:
-                       return BESSConfigUPFBasedIPAllocation()
+                       return BESSConfigUPFBasedIPAllocation(), nil
                }
        }
 
-       panic("Wrong datapath or config type provided")
+       return pfcpiface.Conf{}, fmt.Errorf("Wrong datapath or config type provided")
 }
 
 func PushSliceMeterConfig(sliceConfig pfcpiface.NetworkSlice) error {
diff --git a/test/integration/framework.go b/test/integration/framework.go
index ebaef89..e595708 100644
--- a/test/integration/framework.go
+++ b/test/integration/framework.go
@@ -327,14 +327,19 @@ func setup(t *testing.T, configType uint32) {
                MustStartMockUP4()
        }
 
+       getConfig, err := GetConfig(os.Getenv(EnvDatapath), configType)
+       if err != nil {
+               t.Fatal("Wrong datapath or config type provided")
+       }
+
        switch os.Getenv(EnvMode) {
        case ModeDocker:
-               jsonConf, _ := json.Marshal(GetConfig(os.Getenv(EnvDatapath), configType))
+               jsonConf, _ := json.Marshal(getConfig)
                err := os.WriteFile(ConfigPath, jsonConf, os.ModePerm)
                require.NoError(t, err)
                MustStartPFCPAgent()
        case ModeNative:
-               pfcpAgent = pfcpiface.NewPFCPIface(GetConfig(os.Getenv(EnvDatapath), configType))
+               pfcpAgent = pfcpiface.NewPFCPIface(getConfig)
                go pfcpAgent.Run()
        default:
                t.Fatal("Unexpected test mode")
```

Reply:
@badhrinathpa, any comment on this?